sdks/python/apache_beam/runners/portability/fn_api_runner_test.py

```diff
@@ -306,6 +306,55 @@ def process_timer(self):
       expected = [('fired', ts) for ts in (20, 200)]
       assert_that(actual, equal_to(expected))
 
+  @unittest.skipIf(sys.version_info[0] == 3 and
+                   os.environ.get('RUN_SKIPPED_PY3_TESTS') != '1',
+                   'This test is flaky on on Python 3. '
+                   'TODO: BEAM-5692')
+  def test_pardo_state_timers(self):
+    state_spec = userstate.BagStateSpec('state', beam.coders.StrUtf8Coder())
+    timer_spec = userstate.TimerSpec('timer', userstate.TimeDomain.WATERMARK)
+    elements = list('abcdefgh')
+    buffer_size = 3
+
+    class BufferDoFn(beam.DoFn):
+      def process(self,
+                  kv,
+                  ts=beam.DoFn.TimestampParam,
+                  timer=beam.DoFn.TimerParam(timer_spec),
+                  state=beam.DoFn.StateParam(state_spec)):
+        _, element = kv
+        state.add(element)
+        buffer = state.read()
+        # For reall use, we'd keep track of this size separately.
+        if len(list(buffer)) >= 3:
+          state.clear()
+          yield buffer
+        else:
+          timer.set(ts + 1)
+
+      @userstate.on_timer(timer_spec)
+      def process_timer(self, state=beam.DoFn.StateParam(state_spec)):
+        buffer = state.read()
+        state.clear()
+        yield buffer
+
+    def is_buffered_correctly(actual):
+      # Issues pickling closure of self on jenkins.
```

Comment:
some more detail about this might be useful, unless it would be obvious to someone more familiar with the code